### ISSUE TITLE
Completion on parentheses can take forever ...

### ISSUE THREAD

**Issue by przygienda** [intellij-rust/intellij-rust#2222]
Please provide a knob to switch off the smart behavior on typing ( or ). This can take even in 1000 lines files seconds and makes editing impossible ...

**Comment by Undin** [comment]
If you mean inserting pair brackets, there is platform setting to switch off this behavior
<img width="1024" alt="screen shot 2018-01-23 at 11 19 05" src="https://user-images.githubusercontent.com/2539310/35265038-01fe9d84-0030-11e8-9321-da2884e480b9.png">

**Comment by przygienda** [comment]
ok, I look for taht, same happens when I have large match statements and e.g. put commas behind each case. Blocks for seconds on larg'ish projects/files ...

**Comment by przygienda** [comment]
flipped the knob, seems to fix the issue so far ...

**Comment by przygienda** [comment]
nope, didn't help, issue gets worse and worse with newer releases IMO. The knob doesn't help (is ignored?)
